### PR TITLE
support snapshot get/get_cf

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

**Comment by ngaut** [comment]
Could you squash commit message ?

File: src/rocksdb.rs [hunk 1]
```diff
@@ -446,33 +462,36 @@ impl DB {
...
+    pub fn get(&self, key: &[u8]) -> Result<Option<DBVector>, String> {
+        self.get_opt(key, &ReadOptions::new())
+    }
+
+    pub fn get_cf_opt(&self,
+                      cf: DBCFHandle,
+                      key: &[u8],
+                      readopts: &ReadOptions)
+                      -> Result<Option<DBVector>, String> {
+        if readopts.inner.0.is_null() {
```
Comment (ngaut) [inline src/rocksdb.rs]: Can we use named field instead of number?

Comment (siddontang) [inline src/rocksdb.rs]: no, it uses tuple struct, we can only get it with 0.

File: src/rocksdb.rs [hunk 2]
```diff
@@ -1062,3 +1088,25 @@ mod test {
...
+        assert!(p.is_ok());
+
+        let snap = db.snapshot();
+        let r: Result<Option<DBVector>, String> = snap.get(b"k1");
+        assert!(r.unwrap().unwrap().to_utf8().unwrap() == "v1111");
+
+        let p = db.put(b"k2", b"v2222");
+        assert!(p.is_ok());
+
+        assert!(db.get(b"k2").unwrap().is_some());
```
Comment (ngaut) [inline src/rocksdb.rs]: We should check if we got the same value by using snapshot and db.

Comment (siddontang) [inline src/rocksdb.rs]: no need here, because snapshot mustn't have "k2".

**Comment by ngaut** [comment]
LGTM

**Pull request merged**